### ISSUE TITLE
`z.string().date` is not a function

### ISSUE THREAD

**Issue by debel27** [colinhacks/zod#3387]
I just installed zod version `3.22.4`. 

I'm trying to create a validator for an ISO8061 date string (`YYYY-MM-DD`). According to [the docs](https://zod.dev/?id=iso-date-time-amp-datetime-validation), this can be done as follows:

> The `z.string().date()` method validates strings in the format `YYYY-MM-DD`.

However, the date() method doesn't seem to exist on the object returned by `z.string()`.

## Steps to reproduce

```typescript
const { z } = require("zod")
const dateSchema = z.string().date();
```

My IDE yields a TypeScript error on the above code:

> TS2339: Property `date` does not exist on type `ZodString`

Also, when I run the code, an error is thrown:

> TypeError: z.string(...).date is not a function

This can be verified on runkit: https://runkit.com/embed/ph6clqi91sj2

Screenshot below for reference.

![image](https://github.com/colinhacks/zod/assets/3456755/c92ca6e6-4fe5-4e5b-ae69-ac42ec6b771e)

## Workaround

I'm using the `refine` method as a workaround

```typescript
z.string().refine((str: string) => {
  return (
    /^\d{4}-\d{2}-\d{2}$/.test(str) &&
    new Date(str).toISOString().startsWith(str)
  );
});
```

**Comment by colinhacks** [comment]
Apologies, I merged a PR adding `.date()` but haven't yet released a new version. The docs display the README as it exists in the current `master`.

**Comment by szamanr** [comment]
@colinhacks nice to see `date()` getting added! :tada: any plans to make it customisable? so we could use something like `z.string().dateFormat("dd-MM-yyyy")`?

sorry for writing here, didn't seem like this would warrant opening a new issue.

**Comment by rawand-faraidun** [comment]
@colinhacks any updates on this?

**Comment by debel27** [comment]
> any plans to make it customisable? so we could use something like `z.string().dateFormat("dd-MM-yyyy")`?

Looks like a great idea. But given the current sate of things and how big this feature would be, I think this request should be filed in a separate issue.

**Comment by fwextensions** [comment]
First time using zod in Astro, and just ran into this as well.  Would `z.coerce.date()` be the equivalent for now?

**Comment by cdtut** [comment]
@colinhacks is new release possible with this feature? Lot of good features added since last.

**Comment by hkbertoson** [comment]
> Apologies, I merged a PR adding `.date()` but haven't yet released a new version. The docs display the README as it exists in the current `master`.

I feel like the docs should not have things that have not been released yet.

**Comment by codeams** [comment]
> First time using zod in Astro, and just ran into this as well. Would `z.coerce.date()` be the equivalent for now?

not quite. the output of parsing a schema with coerce.date will be a Date object.

**Comment by colinhacks** [comment]
This is now available in Zod 3.23

**Comment by rodrigoborgesdeoliveira** [comment]
@samarakjackson, it's working for me on v3.23.6.
